File: lib/helldivers_2/models/assignments/assignment/reward.ex
defmodule Helldivers2.Models.Assignments.Assignment.Reward do
  @type t :: %__MODULE__{
    amount: non_neg_integer(),
    id32: non_neg_integer(),
    type: non_neg_integer(),
  }

  defstruct [
    :amount,
    :id32,
    :type,
  ]

  def parse(map) when is_map(map) do
    %__MODULE__{
      amount: Map.get(map, "amount"),
      id32: Map.get(map, "id32"),
      type: Map.get(map, "type"),
    }
  end
end

File: src/Helldivers-2-Models/ArrowHead/Assignments/Reward.cs
namespace Helldivers.Models.ArrowHead.Assignments;

/// <summary>
/// Represents the reward of an <see cref="Assignment" />.
/// </summary>
/// <param name="Type">The type of reward, currently only one value is known: <c>1</c> which represents Medals</param>
/// <param name="Id32">Internal identifier of this <see cref="Reward"/>.</param>
/// <param name="Amount">The amount of <see cref="Type" /> the players will receive upon completion.</param>
public sealed record Reward(
    int Type,
    ulong Id32,
    ulong Amount
);
